### ISSUE TITLE
Material Values Constantly Changing

### ISSUE THREAD

**Issue by HarrisonShort** [kirevdokimov/Unity-UI-Rounded-Corners#11]
Hey @kirevdokimov!

Firstly, thanks for the great plugin, it made designing our in-app catalog a lot easier! It's much appreciated. 
I'm just reaching out to find out if you know why the values of a RoundedCorners material might be changing constantly? 

It seems like every time we open the prefab that has objects using the material, the width, height and radius values change - it's a bit annoying because it constantly shows up in our source control!

Thanks again!

**EDIT:** I've done a bit more testing, it seems like on Play it will be one set of values, and then when opening the prefab (as stated above), it will change. Very weird behaviour! I'm not sure if it's completely specific to your plugin though... 
![image](https://user-images.githubusercontent.com/18225638/92830087-43ac9e00-f418-11ea-9674-2a5816ab321c.png)

**Comment by kirevdokimov** [comment]
Hi @HarrisonShort.
It is changing every time [OnRectTransformDimensionsChange](https://github.com/kirevdokimov/Unity-UI-Rounded-Corners/blob/master/UiRoundedCorners/ImageWithRoundedCorners.cs#L12) or [OnValidate](https://github.com/kirevdokimov/Unity-UI-Rounded-Corners/blob/master/UiRoundedCorners/ImageWithRoundedCorners.cs#L16) is called. Just Unity stuff, nothing special.

Are you sure everything is ok with your layout?

**Comment by Phyrik** [comment]
Hi, I'm getting the same problem here. Every time I open a scene the roundness will change, and when I save the scene, it changes again.

Any help is appreciated, thanks!

**Comment by kirevdokimov** [comment]
Sorry for the late reply @Phyrik. I cannot get this thing so can you or @HarrisonShort send me project (as clear as possible) with bug I can reproduce?

**Comment by HarrisonShort** [comment]
Hey @kirevdokimov, sorry for the very late response, it was a crazy time over the holidays. 
From what I can tell, the material changes on Play when the resolution has been changed, which happens quite a bit as we have devs working on Android and iOS versions of our app on both Windows and Mac computers.

I've attached a sample project that seems to show what is happening. If you select FullContentsButton_RoundedCorners.mat and then play around with resolution size, you should see it changing. You may also have to enter and exit Play mode after changing resolutions.

[MaterialTest.zip](https://github.com/kirevdokimov/Unity-UI-Rounded-Corners/files/5863985/MaterialTest.zip)

**Comment by YigitVeloxia** [comment]
I can confirm that this is happening and it's very frustrating if you are working with VCS. Could you find any temporary fix? @HarrisonShort

**Comment by HarrisonShort** [comment]
@YigitVeloxia Unfortunately I haven't been able to. Currently it seems like the values themselves are not changing anymore... for some reason(?), but the material file still gets marked as if it has changed. Within SourceTree if we select the file, it says "No changes in this file have been detected, or it is a binary file."

We actually added .mat files in the specific folder to the gitignore, but this does not stop changes from appearing as the file has already been committed to the project. Just might explain why the values themselves might not be changing anymore.'

At this stage we've just learned to live with it, but your comment has reminded me to try to see if we can figure out how to stop the issue from happening again. If I come up with anything then I'll post again here.
